core/dnsserver/server-grpc.go

```diff
@@ -20,19 +20,25 @@ import (
 type ServergRPC struct {
 	*Server
 	grpcServer *grpc.Server
-
 	listenAddr net.Addr
+	tlsConfig  *tls.Config
 }
 
 // NewServergRPC returns a new CoreDNS GRPC server and compiles all plugin in to it.
 func NewServergRPC(addr string, group []*Config) (*ServergRPC, error) {
-
 	s, err := NewServer(addr, group)
 	if err != nil {
 		return nil, err
 	}
-	gs := &ServergRPC{Server: s}
-	return gs, nil
+	// The *tls* plugin must make sure that multiple conflicting
+	// TLS configuration return an error: it can only be specified once.
+	var tlsConfig *tls.Config
+	for _, conf := range s.zones {
+		// Should we error if some configs *don't* have TLS?
```

Comment:
FWIW, Caddy does error if there is a conflict in some of the configurations regarding whether to enable TLS.

Reply:
Thanks! Good to know!